raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3347,7 +3357,7 @@ private long pollFollowerAsVoter(FollowerState state, long currentTimeMs) {
         );
     }
 
-    private boolean shouldSendAddOrRemoveVoterRequest(FollowerState state, long currentTimeMs) {
+    private boolean maybeSendAutoJoinRequest(FollowerState state, long currentTimeMs) {
```

Comment:
isn't the method name a little bit misleading? It let me think that it's going to be a "AutoJoin" request (at protocol/API level happening) while it's not the case. Maybe we could name it just as `maybeAutoJoin`?